contracts/truefi2/deprecated/LegacyLoanToken2.sol

```diff
@@ -156,11 +153,14 @@ contract LegacyLoanToken2 is ILoanToken2, ERC20 {
         amount = _amount;
         term = _term;
         apy = _apy;
-        lender = _lender;
         ftlAgency = _ftlAgency;
         debt = interest(amount);
     }
 
+    function setLender(address _lender) external {
```

Comment:
We should guard this just in case we deploy this again.

```suggestion
    function setLender(address _lender) external onlyAdmin {
```

Reply:
This contract has some changes to ease testing, like removing onlyLiquidator modifier. We should not deploy it ever.

Reply:
I see, I misinterpreted the intent of `LegacyLoanToken2` then. If this contract is intended only for use in tests, then let's move this contract to the `mocks/` folder and rename it to `TestLoanToken2`, `TestLegacyLoanToken2`, or something similar. (Can be discussed and done in a separate PR.)

I thought the intent of the `deprecated/` subdir was for contracts that we need to keep for compatibility and in case of urgent upgrade. But if that wasn't the intent then my bad for misinterpreting.